src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -44,6 +48,75 @@ public virtual void Write()
             }
         }
 
+        public virtual void WriteSample(MockTestCase testCase)
+        {
+            generatingSample = true;
+            Regex pattern = new Regex(@".*(Azure.ResourceManager.\w+)/.*");
+            Match match = pattern.Match(Configuration.OutputFolder);
+            if (match.Success)
+            {
+                _writer.additionalComments.Add($"// Need to install {match.Groups[1].Value} v1.0.0 to execute below sample code");
+            }
+            _writer.addUsingRegion = true;
+            _writer.UseNamespace("Azure.ResourceManager");
+            _writer.UseNamespace("System");
+            _writer.UseNamespace("Azure.Identity");
+            _writer.UseNamespace("System.Threading.Tasks");
+
+            _writer.Line($"#region Parameter decalarations");
+            _writer.Line($"// Below parameters are extracted from swagger example files, please use your own values to execute the API lively.");
+            _writer.Line($"ArmClient GetArmClient() => new ArmClient(new DefaultAzureCredential());");
+            //RequestPath requestPath = testCase.Carrier switch
+            //{
+            //    ResourceCollection parentCollection => parentCollection.RequestPath,
+            //    Resource parentResource => parentResource.RequestPath,
+            //    MgmtExtensions parentExtension => parentExtension.ContextualPath,
+            //    _ => throw new InvalidOperationException($"Unknown parent {testCase.Carrier.GetType()}"),
+            //};
+            RequestPath requestPath = testCase.RequestPath;
+            foreach (var fs in testCase.ComposeResourceIdentifierParameterValues(requestPath, true))
```

Comment:
Based on the changes in MockTestCase class, the extra flag `true` changes the meaning of this method, it is returning the serialized names instead of the "parameterValues" as the method names indicated, we need to
- rename the method to make it makes sense in both cases
- change the return value from FormattableString to a more verbose descriptive class or struct to contain both information of the parameter name and parameter value.